tests/report/test_report_items.py

```diff
@@ -801,3 +803,53 @@ def test_3d_caption(cases):
         chart._handle_3d_caption(case=cases[1])
         == "This is case: case-2222222222-2222-2222-2222-2222222222-name with ID: case-2222222222-2222-2222-2222-2222222222"
     )
+
+
+@pytest.mark.usefixtures("mock_detect_latex_compiler")
+def test_subfigure_row_splitting():
+    report_doc = ReportDoc("tester")
+
+    chart = Chart2D(
+        x="nonlinear_residuals/pseudo_step",
+        y="nonlinear_residuals/0_cont",
+        section_title="Continuity convergence",
+        fig_name="convergence_cont",
+        items_in_row=2,
+    )
+
+    chart._add_row_figure(doc=report_doc.doc, img_list=["." for _ in range(6)], fig_caption="test")
+
+    tex = report_doc.doc.dumps()
+
+    lines = tex.split("\n")
+
+    subplots_in_row = 0
+
+    rows = 0
+
+    in_subfigure = False
+    in_figure = False
+
+    caption_in_figure = False
+
+    for line in lines:
+        line = line.lstrip()
+        if line.startswith(r"\caption") and in_figure and (not in_subfigure):
+            caption_in_figure = True
+        if line.startswith(r"\begin{subfigure}[t]"):
```

Comment:
```suggestion
        if line.startswith(r"\begin{subfigure}"):
```

This is purely cosmetic, but is minimally cleaner